mozilla-ai/document-to-podcast
# Add gpu support parler

# What's changing

PR #44 tried to add GPU support for both `text-to-text` and `text-to-speech`, but because it wasn't as straightforward as expected, it was deemed out of scope for now.

This PR simply adds GPU support for the parler models, which was easy to implement.

# How to test it

*Note*:  You need to have a GPU to test it. Maybe a google colab could work?

Steps to test the changes:

1. Make sure you have a CUDA GPU enabled (check with `nvidia-smi` or `torch.cuda.is_available()`)
2. Install the package `pip install -e .`
3. Install the parler dependency `pip install -e '.[parler]'`
4. In the terminal run:
```
document-to-podcast \
--input_file "example_data/Mozilla-Trustworthy_AI.pdf" \
--output_folder "example_data"
--text_to_text_model "Qwen/Qwen2.5-1.5B-Instruct-GGUF/qwen2.5-1.5b-instruct-q8_0.gguf"
--text_to_speech_model "parler-tts/parler-tts-mini-v1"
--device "cuda"
```
5. Check the logs for a message saying "Running parler on cuda" (or similar) (you can also verify your GPU's usage has increased with `nvidia-smi`)
6. You should be able to see a significant speed improvement when generating audio.

# Additional notes for reviewers

# I already...
